File: src/src_core/fmg/core/img/LogoController.java
package fmg.core.img;

/**
 * MVC controller of logo image
 *
 * @param <TImage> platform specific view/image/picture or other display context/canvas/window/panel
 * @param <TImageView> MVC view
 */
public class LogoController<TImage,
                            TImageView extends IImageView<TImage, LogoModel>>
      extends AnimatedImgController<TImage, TImageView, LogoModel>
{

   protected LogoController(TImageView imageView) {
      super(imageView);
   }

   @Override
   public void useRotateTransforming(boolean enable) {
      super.useRotateTransforming(enable);
      if (enable)
         addModelTransformer(new RotateLogoTransformer());
      else
         removeModelTransformer(RotateLogoTransformer.class);
   }

   @Override
   public void usePolarLightFgTransforming(boolean enable) {
      //super.usePolarLightFgTransforming(enable); // hide super implementation!
      if (enable)
         addModelTransformer(new PolarLightLogoTransformer());
      else
         removeModelTransformer(PolarLightLogoTransformer.class);
   }

}

File: src/src_core/fmg/core/img/LogoController.cs
namespace fmg.core.img {

   /// <summary> MVC controlle of logo image </summary>
   /// <typeparam name="TImage">platform specific view/image/picture or other display context/canvas/window/panel</typeparam>
   /// <typeparam name="TImageView">MVC view</typeparam>
   public class LogoController<TImage, TImageView>
       : AnimatedImgController<TImage, TImageView, LogoModel>
      where TImage : class
      where TImageView : IImageView<TImage, LogoModel>
   {

      protected LogoController(TImageView imageView)
         : base(imageView)
      { }

      public override void UseRotateTransforming(bool enable) {
         base.UseRotateTransforming(enable);
         if (enable)
            AddModelTransformer(new RotateLogoTransformer());
         else
            RemoveModelTransformer(typeof(RotateLogoTransformer));
      }

      public override void UsePolarLightFgTransforming(bool enable) {
         //base.UsePolarLightFgTransforming(enable); // hide super implementation!
         if (enable)
            AddModelTransformer(new PolarLightLogoTransformer());
         else
            RemoveModelTransformer(typeof(PolarLightLogoTransformer));
      }

   }

}
